### PR TITLE
Provide common styling for all guides

### PR DESCRIPTION
**Pull request by eriwen**
This injects a header, footer, and a new docs-like theme, and better analytics into the gradle-guides plugin such that all guides who upgrade benefit from this. 

After this change there is no more need for a `docinfo.html` in any guide.

One thing I don't like about the current implementation is that we hardcode the HTML in the plugin class. I tried to load it from a separate file when the plugin is applied, but this seemed equally hacky. Someone might have a clever solution here, though.

screenshot 2018-01-10 11 31 37

### CODE REVIEW

File: src/main/groovy/org/gradle/guides/BasePlugin.groovy [hunk 1]
```diff
@@ -143,10 +145,20 @@ class BasePlugin implements Plugin<Project> {
...
+                    curOut = curOut.replaceAll( ~/<head>/, "<head>${COMMON_HEAD_HTML}")
+
+                    // Inject common header before page title
+                    curOut = curOut.replaceAll( ~/<div id="header">/, """${COMMON_HEADER_HTML}<div id="header">""")
+
+                    // Inject common footer at end of content
+                    curOut = curOut.replaceAll( ~/<\/body>/, """${COMMON_FOOTER_HTML}</body>""")
+
+                    return curOut
                 } else {
```
Comment (bmuschko) [inline src/main/groovy/org/gradle/guides/BasePlugin.groovy]: The `else` is unnecessary. You can just return output.

File: src/main/groovy/org/gradle/guides/BasePlugin.groovy [hunk 2]
```diff
@@ -124,12 +126,12 @@ class BasePlugin implements Plugin<Project> {
...
         }
 
-        addAsciidocExtensions(asciidoc)
+        addAsciidocExtensions(project, asciidoc)
         lazyConfigureMoreAsciidoc(asciidoc)
     }
 
     @CompileDynamic
-    private void addAsciidocExtensions(AsciidoctorTask asciidoc) {
+    private void addAsciidocExtensions(Project project, AsciidoctorTask asciidoc) {
```
Comment (bmuschko) [inline src/main/groovy/org/gradle/guides/BasePlugin.groovy]: Changes should be tested in `BasePluginFunctionalTest.groovy`.

File: src/main/groovy/org/gradle/guides/BasePlugin.groovy [hunk 3]
```diff
@@ -212,4 +224,144 @@ class BasePlugin implements Plugin<Project> {
         }
     }
 
+    // TODO: consider shared-head.html file instead. See http://asciidoctor.org/docs/user-manual/#naming-docinfo-files
+    private static final String COMMON_HEAD_HTML = """
```
Comment (bmuschko) [inline src/main/groovy/org/gradle/guides/BasePlugin.groovy]: I'd rather see this content in external files. For now I'd move them out of the plugin class at least.

Comment (eriwen) [reply src/main/groovy/org/gradle/guides/BasePlugin.groovy]: Me too. Do you have any ideas for this?

Comment (bmuschko) [reply src/main/groovy/org/gradle/guides/BasePlugin.groovy]: Wondering if there's something easy in Asciidoctor. Maybe this helps? https://github.com/asciidoctor/asciidoctor-pdf/issues/89 or the link you posted above. Alternatively, I'd probably just move the content into a file and read it from the plugin.

File: src/main/groovy/org/gradle/guides/BasePlugin.groovy [hunk 4]
```diff
@@ -38,6 +38,9 @@ import org.gradle.api.tasks.PathSensitivity
 @CompileStatic
 class BasePlugin implements Plugin<Project> {
 
+    private static final String COMMON_HEAD_HTML = Thread.currentThread().contextClassLoader.getResource("head-meta.html").text
```
Comment (bmuschko) [inline src/main/groovy/org/gradle/guides/BasePlugin.groovy]: Maybe move this into a method and just pass in the file name.

**Comment by eriwen** [comment]
Thanks @bmuschko merged

**Pull request merged**